Live/Classes/MainScene.cpp

```diff
@@ -98,11 +108,35 @@ void MainScene::update(float delta) {
         isKeyPressed(EventKeyboard::KeyCode::KEY_S)) {
         m_player->moveY(-(MOVE_STEP * delta));
     }
+    if (isKeyPressed(EventKeyboard::KeyCode::KEY_Z)) {
+        for (auto it : m_items) {
+            if (m_player->pickup(it)) {
+                m_items.erase(std::remove(m_items.begin(), m_items.end(), it));
```

Comment:
if you use iterators instead, you can do `m_items.erase(iter)`. And also use an `std::set` instead, it's going to be faster and more appropriate. For the loop, do `for (auto iter  = m_items.begin(); iter != m_items.end(); iter++)` and change to `m_player->pickup(*iter)` and `m_items.erase(iter);`.